crates/stages/src/stages/sender_recovery.rs

```diff
@@ -91,24 +77,54 @@ impl<DB: Database> Stage<DB> for SenderRecoveryStage {
 
         // Iterate over transactions in chunks
         info!(target: "sync::stages::sender_recovery", start_tx_index, end_tx_index, "Recovering senders");
-        for chunk in &entries.chunks(self.batch_size) {
-            let transactions = chunk.collect::<Result<Vec<_>, DbError>>()?;
-            // Recover signers for the chunk in parallel
-            let recovered = transactions
-                .into_par_iter()
-                .map(|(tx_id, transaction)| {
-                    trace!(target: "sync::stages::sender_recovery", tx_id, hash = ?transaction.hash(), "Recovering sender");
-                    let signer =
-                        transaction.recover_signer().ok_or_else::<StageError, _>(|| {
-                            SenderRecoveryStageError::SenderRecovery { tx: tx_id }.into()
-                        })?;
+
+        // a channel to receive results from a rayon job
+        let (tx, rx) = mpsc::unbounded_channel();
+
+        // spawn recovery jobs onto the default rayon threadpool and send the result through the
+        // channel
+        for entry in entries {
+            let (tx_id, transaction) = entry?;
+            let tx = tx.clone();
+            rayon::spawn(move || {
```

Comment:
nit to prioritize earlier senders
```suggestion
            rayon::spawn_fifo(move || {
```